src/main/java/bowling/domain/frame/Frames.java

```diff
@@ -0,0 +1,78 @@
+package bowling.domain.frame;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.IntStream;
+
+public final class Frames {
+    private final List<Frame> frames;
+
+    public Frames() {
+        this(generateFrames());
+    }
+
+    public Frames(List<Frame> frames) {
+        this.frames = frames;
+    }
+
+    public Frames addScore(int score) throws Exception {
+        if (isFinished()) {
+            throw new IndexOutOfBoundsException();
+        }
+        this.frames.get(nowFrame()).addScore(score);
+        return new Frames(this.frames);
+    }
+
+    public boolean isFinished() {
+        if (nowFrame() > 9) {
+            return true;
+        }
+        return false;
```

Comment:
```suggestion
        return nowFrame() > 9;
```
이렇게 표현할 수 있습니다.
매직넘버(9)는 상수로 표현해주면 어떨까요?